src/api/tasksService.js

```diff
@@ -0,0 +1,350 @@
+    throw err;
+  }
+};
+
+/**
+ * updateSubtask - updates a subtask
+ * @param {Number} taskId Id of task of which subtask is to be updated
+ * @param {Number} subtaskId Id of subtask to be updated
+ * @param {object} subtaskData subtask update data
+ * @returns returns the task with updated subtask
+ */
+export const updateSubtask = async (taskId, subtaskId, subtaskData) => {
+  try {
+    //get task
+    const res = await api.get(`/tasks/${taskId}`);
+    const task = res.data;
+    if (!task) {
+      throw new Error("updateSubtask: Task not found");
+    }
+
+    //get Index of Subtask
+    const subtaskIndex = task.subtasks.findIndex((st) => st.id === subtaskId);
+    if (subtaskIndex === -1) {
+      throw new Error("updateSubtask: Subtask not found");
+    }
+
+    //strip the id off subtask data if present
+    const subtaskDataWithoutId = removeId(subtaskData);
+
+    //update specific subtask
+    task.subtasks[subtaskIndex] = {...task.subtasks[subtaskIndex], ...subtaskDataWithoutId};
+
+    //patch data and save to db
+    const updatedTask = await api.patch(`/tasks/${taskId}`, task);
+    return updatedTask.data;
+  } catch (err) {
+    console.error("Error updating subtask:", err);
+    throw err;
+  }
+};
+
+//DELETE ENDPOINTS
+/**
+ * deleteTask - deletes a task
+ * @param {Number} taskId ID of the task to be deleted
+ * @returns {object} returns a success status and message
+ */
+export const deleteTask = async (taskId) => {
+  try {
+    await api.delete(`/tasks/${taskId}`);
+    return {
+      status: "success",
+      message: "Task deleted successfully",
+    };
+  } catch (err) {
+    console.error("Error deleting task: ", err);
+    throw err;
+  }
+};
+
+/**
+ * deleteSubtask - deletes a subtask
+ * @param {Number} taskId ID of task which contains subtask to be deleted
+ * @param {Number} subtaskId ID of subtask to be deleted
+ * @returns {object} An updated data with subtask deleted
+ */
+export const deleteSubtask = async (taskId, subtaskId) => {
+  try {
+    //get Task
+    const res = await api.get(`/tasks/${taskId}`);
+    const task = res.data;
+
+    if (!task) {
+      throw new Error("deleteSubtask: Task not found");
+    }
+    const subtaskIndex = task.subtasks.findIndex((st) => st.id === subtaskId);
+    if (subtaskIndex === -1) {
+      throw new Error("deleteSubtask: Subtask not found");
+    }
+    //remove the subtask at this index
+    task.subtasks.splice(subtaskIndex, 1);
+    const updatedTask = await api.patch(`/tasks/${taskId}`, task);
+    return updatedTask.data;
+  } catch (err) {
+    console.error("Error deleting subtask: ", err);
+    throw err;
+  }
+};
+
+// /**
+//  * deleteNote - deletes a note
+//  * @param {Number} taskId ID of task which contains note to be deleted
+//  * @param {Number} noteId ID of note to be deleted
+//  * @returns {object} An updated data with note deleted
+//  */
+
+// export const deleteNote = async (taskId, noteId) => {
```

Comment:
@Haydox01  suggest that the schema has the appropraite ids now. Please test this feature as well